### PR TITLE
[FIX] mail: fix new message separator of unknown message

### PR DESCRIPTION
**Pull request by tsm-odoo**
Before this PR, the new message separator was put on top of the first known message when the message was not known. This PR fixes the issue.

Steps to reproduce the issue:
- post 100 message on the general channel
- mark the first message as unread
- reload the page
- note that the separator is placed on top of the first loaded message.

### CODE REVIEW

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

File: addons/mail/static/src/core/common/thread.js [hunk 1]
```diff
@@ -454,24 +454,20 @@ export class Thread extends Component {
...
         const separator = this.props.thread.selfMember?.new_message_separator;
-        return (
-            (separator === 0 && isFirst) ||
-            current.id === separator ||
-            (current.id > separator && (!previous || previous?.id < separator))
-        );
+        if (current.id === separator) {
+            return true;
+        }
+        if (current.id <= separator) {
```
Comment (seb-odoo) [inline addons/mail/static/src/core/common/thread.js:468]: `<=` should be `<` here, it makes no difference due to ordering of condition but having `=` says the opposite of the first condition.

Comment (alexkuhn) [reply addons/mail/static/src/core/common/thread.js]: done!

**Comment by alexkuhn** [comment]
@robodoo r+ 😏

**Pull request closed without merging**